R/add_weights.R

```diff
@@ -0,0 +1,70 @@
+#' Add a weight variable using the sample frame
+#'
+#' @param .dataset the clean dataframe
+#' @param sample_data sample dataframe including poplution numbers and the strata
+#' @param strata_column_dataset name of strata column in the clean dataframe
+#' @param strata_column_sample name of strata column in the sample dataframe
+#' @param population_column name of population column in the sample dataframe
+#' @param weight_column name of the added weight column. By default "weight"
+#'
+#' @return The clean dataset with 1 new column: weight
+#' @export
+#'
+#' @examples
+#' clean_data <- data.frame(uuid = c(1,2,3,4,5,6,7,8),
+#'                          strata = c("strata1","strata2","strata1",
+#'                                     "strata2","strata1","strata2",
+#'                                     "strata1","strata1"))
+#' sample <- data.frame(strata = c("strata1","strata2"),
+#'                      population = c(30000,50000))
+#'
+#' clean_data_weighted <- clean_data %>%
+#'     add_weights(sample,
+#'                 strata_column_dataset = "strata",
+#'                 strata_column_sample = "strata",
+#'                 population_column = "population")
+add_weights<- function(.dataset,
+                        sample_data,
+                        strata_column_dataset = NULL,
+                        strata_column_sample = NULL,
+                        population_column = NULL,
+                        weight_column = "weight"){
```

Comment:
if time allows, can we change weight_column = "weight", to "weights" to match the name of the function.